mrmustard/utils/parametrized.py

```diff
@@ -31,73 +32,28 @@ class Parametrized:
     """
 
     def __init__(self, **kwargs):  # NOTE: only kwargs so that we can use the arg names
-        self._trainable_parameters = {"symplectic": [], "euclidean": [], "orthogonal": []}
-        self._constant_parameters = {"symplectic": [], "euclidean": [], "orthogonal": []}
-        self._param_names = []
+
         owner = f"{self.__class__.__qualname__}"
+
         for name, value in kwargs.items():
-            trainable = (name + "_trainable" in kwargs and kwargs[name + "_trainable"] is True) or (
-                math.from_backend(value) and math.is_trainable(value)
-            )
-            constant = (
-                name + "_trainable" in kwargs
-                and kwargs[name + "_trainable"] is False
-                or (math.from_backend(value) and not math.is_trainable(value))
-            )
-            param_type = (
-                "symplectic"
-                if name.startswith("symplectic")
-                else "orthogonal"
-                if name.startswith("orthogonal")
-                else "euclidean"
-            )
-            if trainable:
-                if not math.from_backend(value) or (
-                    math.from_backend(value) and not math.is_trainable(value)
-                ):
-                    value = math.new_variable(value, kwargs[name + "_bounds"], owner + ":" + name)
-                self._trainable_parameters[param_type].append(value)
-            elif constant:
-                if not math.from_backend(value):
-                    value = math.new_constant(value, owner + ":" + name)
-                self._constant_parameters[param_type].append(value)
-            else:
-                name = "_" + name
-            self.__dict__[name] = value
-            self._param_names += [] if name.startswith("_") else [name]
+
+            # filter `{name}_trainable` or `{name}_bounds`` to become fields
+            # of the class as those kwargs are used to define the variables
+            if "_trainable" in name or "_bounds" in name:
+                continue
+
+            # convert into parameter class
+            is_trainable = kwargs.get(f"{name}_trainable", False) or math.is_trainable(value)
+            bounds = kwargs.get(f"{name}_bounds", None)
+            param_value = create_parameter(value, name, is_trainable, bounds, owner)
+
+            # dynamically assign variable as attribute of the class
+            self.__dict__[f"{name}"] = param_value
```

Comment:
why not just `[name]` if it's already a string?

Reply:
Sure! before it was something like `self.__dict__[f"_{name}"]` and then forgot to change it 🙈